src/m365/spo/commands/folder/folder-add.spec.ts

```diff
@@ -194,6 +197,44 @@ describe(commands.FOLDER_ADD, () => {
     });
   });
 
+  it('creates folder and ensures parent folder path', async () => {
+    const requestGet = sinon.stub(request, 'get').callsFake(async (opts) => {
+      if ((opts.url as string).indexOf('GetFolderByServerRelativePath(') > -1) {
+        return {
+          Exists: false
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    const requestPost: sinon.SinonStub = stubPostResponses();
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: 'https://contoso.sharepoint.com',
+        parentFolderUrl: '/Shared Documents/abc',
+        name: 'folder1',
+        ensureParentFolders: true
+      }
+    });
+
+    assert.strictEqual(requestGet.lastCall.args[0].url, 'https://contoso.sharepoint.com/_api/web/GetFolderByServerRelativePath(DecodedUrl=\'Shared%20Documents%2Fabc\')?$select=Exists');
```

Comment:
Let's also encode the shared%20documents ... here. Also let's use backticks instead of escaping the '
```suggestion
    assert.strictEqual(requestGet.lastCall.args[0].url, `https://contoso.sharepoint.com/_api/web/GetFolderByServerRelativePath(DecodedUrl='${formatting.encodeQueryParameter(...)}')?$select=Exists`);
```